validator/src/bin/solana-test-validator.rs

```diff
@@ -0,0 +1,263 @@
+use {
+    clap::{value_t_or_exit, App, Arg},
+    console::style,
+    indicatif::{ProgressBar, ProgressDrawTarget, ProgressStyle},
+    solana_clap_utils::{input_parsers::pubkey_of, input_validators::is_pubkey},
+    solana_client::{client_error, rpc_client::RpcClient},
+    solana_core::rpc::JsonRpcConfig,
+    solana_sdk::{
+        clock::{Slot, DEFAULT_TICKS_PER_SLOT, MS_PER_TICK},
+        commitment_config::CommitmentConfig,
+        fee_calculator::FeeRateGovernor,
+        rent::Rent,
+        rpc_port,
+        signature::{read_keypair_file, Signer},
+    },
+    solana_validator::{start_logger, test_validator::*},
+    std::{
+        path::PathBuf,
+        process::exit,
+        thread::sleep,
+        time::{Duration, SystemTime, UNIX_EPOCH},
+    },
+};
+
+#[derive(PartialEq)]
+enum Output {
+    None,
+    Log,
+    Dashboard,
+}
+
+/// Creates a new process bar for processing that will take an unknown amount of time
+fn new_spinner_progress_bar() -> ProgressBar {
+    let progress_bar = ProgressBar::new(42);
+    progress_bar.set_draw_target(ProgressDrawTarget::stdout());
+    progress_bar
+        .set_style(ProgressStyle::default_spinner().template("{spinner:.green} {wide_msg}"));
+    progress_bar.enable_steady_tick(100);
+    progress_bar
+}
+
+/// Pretty print a "name value"
+fn println_name_value(name: &str, value: &str) {
+    println!("{} {}", style(name).bold(), value);
+}
+
+fn main() {
+    let default_rpc_port = rpc_port::DEFAULT_RPC_PORT.to_string();
+
+    let matches = App::new("solana-test-validator").about("Test Validator")
+        .version(solana_version::version!())
+        .arg({
+            let arg = Arg::with_name("config_file")
+                .short("C")
+                .long("config")
+                .value_name("PATH")
+                .takes_value(true)
+                .help("Configuration file to use");
+            if let Some(ref config_file) = *solana_cli_config::CONFIG_FILE {
+                arg.default_value(&config_file)
+            } else {
+                arg
+            }
+        })
+        .arg(
+            Arg::with_name("mint_address")
+                .long("mint")
+                .value_name("PUBKEY")
+                .validator(is_pubkey)
+                .takes_value(true)
+                .help("Address of the mint account that will receive all the initial tokens [default: client keypair]"),
+        )
+        .arg(
+            Arg::with_name("ledger_path")
+                .short("l")
+                .long("ledger")
+                .value_name("DIR")
+                .takes_value(true)
+                .required(true)
+                .default_value("test-ledger")
+                .help("Use DIR as ledger location"),
```

Comment:
What do you think about an arg like `--clean` or some such that would empty the ledger dir before initializing? I find I often want to start fresh.

Reply:
I've been thinking about that and going back and forth.  I finally concluded that adding "clean" support didn't really add much, and I'd rather just `rm -rf test-ledger` myself from the shell/script before running `solana-test-validator`.  I think I could be convinced otherwise though